lib/assessment.js

```diff
@@ -32,14 +48,27 @@ var any = function any(array, callback, thisArg) {
 // made up of the array elements for which invoking
 // the callback returns a truthy value
 var select = function select(array, callback, thisArg) {
-
+  var filtered = [];
+  // Iterate through each element in an array
+  array.forEach(function(value) {
+    // Test each value by passing it to the predicate
+    if(callback(value)) {
+      // If the predicate returns true, add it to the new array
+      filtered.push(value);
+    }
+  });
+  // Return the filtered array
+  return filtered;
 };
 
 // Question 5
 // Write a function that returns the first element
 // for which invoking the callback returns a truthy value
 var find = function find(array, callback, thisArg) {
-
+  // Collect all truthy values in an array
+  var truthy_values = select(array, callback);
+  // Return the first element of the array
+  return truthy_values[0];
```

Comment:
Good. You saw that this could be implemented using code you had already written. 1